File: src/01-easy/temperatures/solution.vb
Module Answer
    Sub Main()
        Dim n = CInt(Console.ReadLine())
        If n = 0 Then
            Console.WriteLine(0)
            Return
        End If
        Dim t = Console.ReadLine().Split(" "c)
        Dim r = CInt(t(0))
        For Each s In t
            Dim v = CInt(s)
            If Math.Abs(v) < Math.Abs(r) OrElse (Math.Abs(v) = Math.Abs(r) AndAlso v > 0) Then r = v
        Next
        Console.WriteLine(r)
    End Sub
End Module

File: src/01-easy/temperatures/solution.cs
using System;
using System.Linq;
class Solution {
    static void Main() {
        int n = int.Parse(Console.ReadLine());
        if (n == 0) { Console.WriteLine(0); return; }
        var t = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        int r = t[0];
        foreach (var v in t)
            if (Math.Abs(v) < Math.Abs(r) || (Math.Abs(v) == Math.Abs(r) && v > 0)) r = v;
        Console.WriteLine(r);
    }
}
